src/pages/calendar.tsx

```diff
@@ -1,32 +1,398 @@
-import { Calendar as ReactCalendar } from 'react-calendar';
+import { useState, useEffect } from 'react';
+import dynamic from 'next/dynamic';
 import 'react-calendar/dist/Calendar.css';
+import {
+  Box,
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  Select,
+  Text,
+  VStack,
+  Flex,
+} from '@chakra-ui/react';
+import InputForm from '../components/InputForm';
 
-import { Container, Text } from '@chakra-ui/react';
-
-import React, { useState } from 'react';
+const Calendar = dynamic(() => import('react-calendar'), { ssr: false });
 
 type ValuePiece = Date | null;
-
 type Value = ValuePiece | [ValuePiece, ValuePiece];
 
-const Calendar = () => {
-  const [value, onChange] = useState<Value>(new Date());
-  const [clickedDate, setClickedDate] = useState<ValuePiece>(null);
+interface Event {
+  id: number;
+  title: string;
+  lessonDate: Date;
+  subject: string;
+  lessonPlan: string | null;
+  reasonOfAbsence: string;
+  absentTeacherId: number;
+  substituteTeacherId: number | null;
+  locationId: number;
+  newAbsence?: Omit<Event, 'id'>;
+}
+
+interface FetchEventsResponse {
+  absences: Event[];
+}
+
+function CalendarView() {
+  const [events, setEvents] = useState<Event[]>([]);
+  const [value, setValue] = useState<Value>(new Date());
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [formDate, setFormDate] = useState<Date | null>(null);
+  const [view, setView] = useState('month');
+
+  useEffect(() => {
+    FetchEvents();
+  }, []);
+
+  const FetchEvents = async () => {
```

Comment:
i think this should fetch from the db based on if its in day, week, or month mode. I feel like this would be better than grabbing all absences in the db, especially if there are be lot of absences. The useEffect can run this function every time the view is changed.

Reply:
but we anyways have to fetch them initially for month view in the beginning so won't it be unnecessary to fetch multiple times for different views